src/gt4py/next/ffront/fbuiltins.py

```diff
@@ -29,6 +30,12 @@
 TYPE_BUILTINS = [Field, Dimension, int32, int64, float32, float64] + PYTHON_TYPE_BUILTINS
 TYPE_BUILTIN_NAMES = [t.__name__ for t in TYPE_BUILTINS]
 
+# Be aware: Type aliases are not fully supported in the frontend yet, e.g. `IndexType(1)` will not
+# work.
+IndexType: TypeAlias = int32
```

Comment:
What was the problem why we decided to add an index type? I only realize now that this is a potential problem for bindings, if the driver code has an opinion on the size of `IndexType`. But let's keep it for now...

Reply:
You suggested it as it is occurring in the tests multiple times (e.g. in IndexFields). I think the bindings will still accept `int64` for NeighborTableOffsetProvider.

Reply:
I suggested the alias but why did we introduce this constraint https://github.com/GridTools/gt4py/pull/1255/files#diff-1c4cbeb95e161c4e1ca0327817608ba3a8b48569e756847b2093bc43606863d6R685